rust/main/agents/relayer/src/msg/metadata/multisig/merkle_root_multisig.rs

```diff
@@ -36,19 +36,48 @@ impl MultisigIsmMetadataBuilder for MerkleRootMultisigMetadataBuilder {
         checkpoint_syncer: &MultisigCheckpointSyncer,
     ) -> Result<Option<MultisigMetadata>> {
         const CTX: &str = "When fetching MerkleRootMultisig metadata";
-        let highest_leaf_index = unwrap_or_none_result!(
-            self.highest_known_leaf_index().await,
-            debug!("Couldn't get highest known leaf index")
-        );
-        let leaf_index = unwrap_or_none_result!(
-            self.get_merkle_leaf_id_by_message_id(message.id())
-                .await
-                .context(CTX)?,
-            debug!(
-                hyp_message=?message,
-                "No merkle leaf found for message id, must have not been enqueued in the tree"
-            )
-        );
+
+        let fn_name = "highest_known_leaf_index";
+        let highest_leaf_index = match self
+            .get_cached_call_result::<u32>(None, fn_name, &NoParams)
+            .await
+        {
+            Some(index) => index,
+            None => {
+                let index: u32 = unwrap_or_none_result!(
+                    self.highest_known_leaf_index().await,
+                    debug!("Couldn't get highest known leaf index")
+                );
+
+                self.cache_call_result(None, fn_name, &NoParams, &index)
+                    .await;
+                index
+            }
+        };
```

Comment:
I would like to see this expansions of making a call directly to RPC with support for cached results to be extracted into an individual method. It contains a hidden contract on value of `fn_name` to be the same as RPC method name which can be explained in the method doc comment.

Reply:
As discussed, moved the calls to their own functions to aid readability.
As a bonus, moved all the other cached ism calls as well. 

[Feat](https://github.com/Mantas-M/hyperlane-monorepo/commit/d7767694d9314029e47a55edcec261f8ca5ef12e)

Reply:
I added a doc comment to one of such methods. It will be great if similar comments are added to all the methods.